File: TypedActorExample/src/main/java/org/akka/essentials/calculator/CalculatorInt.java
package org.akka.essentials.calculator;

import akka.dispatch.Future;
import akka.japi.Option;

public interface CalculatorInt {

	public Future<Integer> add(Integer first, Integer second);

	public Future<Integer> subtract(Integer first, Integer second);

	public void incrementCount();

	public Option<Integer> incrementAndReturn();

}

File: TypedActorExample/src/main/scala/org/akka/essentials/calculator/CalculatorInt.scala
package org.akka.essentials.calculator
import akka.dispatch.Future
import akka.actor.TypedActor.Receiver
import akka.actor.TypedActor.PreStart
import akka.actor.TypedActor.Supervisor
import akka.actor.TypedActor.PostStop


trait CalculatorInt extends Receiver  {

	def add(first: Int, second: Int): Future[Int]

	def subtract(first: Int, second: Int): Future[Int]

	def incrementCount(): Unit

	def incrementAndReturn(): Option[Int]
}
